modules/mbedtls/Kconfig.tls-generic

```diff
@@ -559,6 +559,16 @@ config MBEDTLS_PSA_P256M_DRIVER_ENABLED
 	  Enable support for the optimized sofware implementation of the secp256r1
 	  curve through the standard PSA API.
 
+config MBEDTLS_PSA_KEY_SLOT_COUNT
+	int "Number of key slots in PSA core"
+	default 32
```

Comment:
Well, as of Mbed TLS 3.6.2 you have 3 options:

1) legacy - what was already there up to 3.6.0

This is when `!MBEDTLS_PSA_KEY_STORE_DYNAMIC && !MBEDTLS_PSA_STATIC_KEY_SLOTS`. In this case you have `MBEDTLS_PSA_KEY_SLOT_COUNT` slots available, each of which has the key material stored on heap whenever that slot starts to be used.

2) fully static - what I recently added in Mbed TLS and that will be available in 3.6.3 or through the patch I'm introducing with this PR.

This is when `!MBEDTLS_PSA_KEY_STORE_DYNAMIC && MBEDTLS_PSA_STATIC_KEY_SLOTS`. In this case you have `MBEDTLS_PSA_KEY_SLOT_COUNT` slots available, each of which has the key material stored on a static slot allocated at build time.

3) fully dynamic

This is when `MBEDTLS_PSA_KEY_STORE_DYNAMIC` is set. In this case everything is on the heap memory and it's allocated when required. In this case the number of available slots is [2^25](https://github.com/Mbed-TLS/mbedtls/blob/mbedtls-3.6/docs/architecture/psa-keystore-design.md#length-of-the-volatile-key-slices) which basically means you'll run out of heap memory way before that.

Reply:
Sorry I forgot to answer your question in the end.

> Do you happen to know what was the maximum key slot count when using the dynamic version?

It was 32, because we didn't set that explicitly in Zephyr (a part from 1 specific case with BT mesh), we were using the Mbed TLS default. So I think that no, this should not break stuff as we're setting what it was already done before implicitly.